### PR TITLE
[WFLY-14947] Create analysis doc for Micrometer Support

### PR DESCRIPTION
**Pull request by jasondlee**
https://issues.redhat.com/browse/WFLY-14947

Create analysis doc

### CODE REVIEW

File: observability/WFLY-14947-Implement_observability_policy_metrics.adoc [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+=== Hard Requirements
+* A single registry for metrics
+* Server and JVM metrics tracked and exposed
+* Availability of metric registry for injection into user applications
+* Endpoint to expose metrics in Prometheus format
+* RBAC support for server/jvm metrics
+* Remove base/vendor prefixes (in accordance with Prometheus standards)
+* This subsystem will replace the existing WildFly Metrics subsystem. Documentation will be provided describing the change.
+* If MicroProfile Metrics is enabled, this subsystem will decline to install itself, as is the current behavior of WildFly Metrics.
```
Comment (bstansberry) [inline observability/WFLY-14947-Implement_observability_policy_metrics.adoc:50]: @jasondlee We need migration requirements for the existing WildFly Metrics subsystem.

Either:

1) It continues to work but there is appropriate logic to deal with the case where both it and the Micrometer subsystem are in a non-admin-only server config.
2) It becomes a model-only / admin-only subsystem with a migrate op added to migrate to the new subsystem.

File: observability/WFLY-14947-Implement_observability_policy_metrics.adoc [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+=== Nice-to-Have Requirements
+* Automatic timing of REST requests
+* Mechanism for arbitrary subsystems to register meters (e.g., Micrometer's `MeterBinder`)
```
Comment (bstansberry) [inline observability/WFLY-14947-Implement_observability_policy_metrics.adoc:54]: At this stage I think these need to either be Hard Requirements or Non-Requirements. Nice-to-Have Requirements are ok for a while during earlier dev work but eventually should be committed or not committed.

A Non-Requirement can always be added in a later RFE.

File: observability/WFLY-14947-Implement_observability_policy_metrics.adoc [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+:author:            Jason Lee
+:email:             jasondlee@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+While WildFly/EAP already supports basic metrics in the base WildFly/EAP system, as well as MicroProfile Metrics in the WildFly MicroProfile configuration and EAP XP, there is an effort in Red Hat Middleware Runtimes to implement a consistent observability policy across all Runtime offerings. In the Java space, the library of choice, and the emerging de facto standard for Java developers is https://micrometer.io[Micrometer].  This RFE will add support for Micrometer to WildFly/EAP and adhere to the policy decided upon by the various stakeholders.
```
Comment (bstansberry) [inline observability/WFLY-14947-Implement_observability_policy_metrics.adoc:11]: This is a WildFly community document, except for the EAP7 JIRA link. :)

Comment (jasondlee) [reply observability/WFLY-14947-Implement_observability_policy_metrics.adoc]: EAP references removed (except for the link)

File: observability/WFLY-14947-Implement_observability_policy_metrics.adoc (hunk 2 above)
Comment (jasondlee) [reply observability/WFLY-14947-Implement_observability_policy_metrics.adoc]: Moved items to `Non-requirements` section.

File: observability/WFLY-14947-Implement_observability_policy_metrics.adoc (hunk 1 above)
Comment (jasondlee) [reply observability/WFLY-14947-Implement_observability_policy_metrics.adoc]: I've update the language here a bit, but I'm not entirely sure what more you're looking for if this is insufficient.

File: observability/WFLY-14947-Implement_observability_policy_metrics.adoc [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+While WildFly already supports basic metrics in the base WildFly system, as well as MicroProfile Metrics in the WildFly MicroProfile configuration, there is an effort in Red Hat Middleware Runtimes to implement a consistent observability policy across all Runtime offerings. In the Java space, the library of choice, and the emerging de facto standard for Java developers is https://micrometer.io[Micrometer].  This RFE will add support for Micrometer to WildFly Preview and adhere to the policy decided upon by the various stakeholders.
+
+== Issue Metadata
+
+=== Issue
+* https://issues.redhat.com/browse/WFLY-14947
+
+=== Related Issues
+* https://issues.redhat.com/browse7-1686
```
Comment (bstansberry) [inline observability/WFLY-14947-Implement_observability_policy_metrics.adoc:19]: This is a broken URL.

**Pull request merged**